extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/photos/MicrosoftPhotosImporter.java

```diff
@@ -187,6 +192,17 @@ private String importSinglePhoto(
         if (code < 200 || code > 299) {
           throw new IOException("Got error code: " + code + " message " + response.message());
         }
+        if (code == 401){
+            // If there was an unauthorized error, then try refreshing the creds
+            credentialFactory.refreshCredential(credentialFactory.createCredential(authData));
+            monitor.info(() -> "Refreshed authorization token successfuly");
+
+            Response newResponse = client.newCall(requestBuilder.build()).execute();
```

Comment:
Is the new token attached here? Or do we have to recreate the request?

Reply:
FYI Will did this with a Supplier in #777 which is pretty neat

Reply:
I believe it's attached: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#successful-response-2

Should I do something with it? just add it to the AuthData field?

Reply:
Oh sorry I meant, is it attched to the requestBuilder? I think it may have the older invalid credential and needs to be rebuilt. 

The older cred is added on line 181:  `requestBuilder.header("Authorization", "Bearer " + authData.getAccessToken()); ` I think we just need to replace the header.